test2/2.2/stack.c

```diff
@@ -0,0 +1,57 @@
+#include <malloc.h>
+#include "stack.h"
+
+int push(Node **head, const int value) {
+    Node *newNode = (Node *) malloc(sizeof(Node));
+    if (newNode == NULL || head == NULL) {
+        return 1;
+    }
+    newNode->value = value;
+    newNode->next = (*head);
+    (*head) = newNode;
```

Comment:
```suggestion
    *head = newNode;
```
И ещё много где, дальше отмечать не буду.